libraries/cyclestreets-view/src/test/java/net/cyclestreets/liveride/LiveRideTest.kt

```diff
@@ -0,0 +1,120 @@
+    private val mockCycleStreetsNotifications = mock(CycleStreetsNotifications::class.java)
+    private val mockNotificationBuilder = mock(Notification.Builder::class.java)
+    private val mockNotification = mock(Notification::class.java)
+
+    @Before
+    fun setUp() {
+        setUpMocks()
+        CycleStreetsPreferences.initialise(roboContext, -1)
+
+        liveRideState = LiveRideStart(roboContext, mockTts, mockCycleStreetsNotifications)
+        liveRideState = OnTheMove(liveRideState)
+        verify("Starting Live Ride")
+
+        Route.initialise(roboContext)
+    }
+
+    private fun setUpMocks() {
+        `when`(mockCycleStreetsNotifications.builder).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.setSmallIcon(anyInt())).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.setTicker(anyString())).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.setWhen(anyLong())).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.setAutoCancel(anyBoolean())).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.setOngoing(anyBoolean())).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.setContentTitle(any())).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.setContentText(any())).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.setContentIntent(any())).thenReturn(mockNotificationBuilder)
+        `when`(mockNotificationBuilder.build()).thenReturn(mockNotification)
+    }
+
+    @Test
+    fun doubleStraightOn() {
+        loadJourneyFrom("journey-domain.json")
+        journey.setActiveSegmentIndex(13)
+        assertThat(journey.activeSegment().street()).isEqualTo("lcn (unknown cycle network)")
+
+        move(0.126, 52.212)
+        verify("Get ready to Straight on")
+        move(0.12634, 52.21207)
+        verify("Straight on into lcn (unknown cycle network) continuation. Continue 5m")
+        move(0.12634, 52.21207)
+        // AdvanceToSegment advances the segment; we need another location update to trigger the next NearingTurn
+        move(0.12634, 52.21207)
+        verify("Get ready to Straight on")
+        move(0.12634, 52.21207)
+        verify("Straight on into lcn (unknown cycle network). Continue 85m")
+    }
+
+    @Test
+    fun specificTurnIssues() {
+        loadJourneyFrom("journey-rightleft-domain.json")
+        journey.setActiveSegmentIndex(1)
+        assertThat(journey.activeSegment().street()).isEqualTo("Link with A38")
+
+        move(-3.33022, 50.92086)
+        verify("Get ready to Turn right")
+        move(-3.33019, 50.92081)
+        verify("Turn right into A38. Continue 15m")
+        move(-3.33019, 50.92081)
+        move(-3.33019, 50.92081)
+        verify("Get ready to Turn left")
+        move(-3.33019, 50.92081)
+        verify("Turn left into Broad Path. Continue 990m")
```

Comment:
These 10 lines above are the interesting part of this PR.

The key point is that it's not until you hit the junction itself and get the "Continue 15m" that you get any clue that you're not going straight across.  If you're cycling quickly on the approach to the junction (if visibility is good), then you could be 30 metres past the turning into Broad Path before you realise your error.